hassio/src/dialogs/datadisk/dialog-hassio-datadisk.ts

```diff
@@ -0,0 +1,149 @@
+  (hardware: HassioHardwareInfo, current_path: string) =>
+    hardware.devices
+      .filter(
+        (device) =>
+          device.attributes?.DEVTYPE === "disk" &&
+          device.attributes?.MINOR === "0" &&
+          !device.dev_path.startsWith(current_path)
+      )
+      .sort((a, b) => stringCompare(a.name, b.name))
+);
+
+@customElement("dialog-hassio-datadisk")
+class HassioDatadiskDialog extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _params?: HassioDatatiskDialogParams;
+
+  @state() private _selected_device?: string;
+
+  public showDialog(params: HassioDatatiskDialogParams) {
+    this._params = params;
+  }
+
+  public closeDialog(): void {
+    this._params = undefined;
+    this._selected_device = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params) {
+      return html``;
+    }
+    const devices = _filterDevices(
+      this._params.hardware,
+      this._params.supervisor.os.data_disk
+    );
+    return html`
+      <ha-dialog
+        open
+        scrimClickAction
+        escapeKeyAction
+        @closed=${this.closeDialog}
+        .heading=${createCloseHeading(
+          this.hass,
+          this._params.supervisor.localize("dialog.datadisk_move.title")
+        )}
+      >
+        ${devices.length
+          ? html`
+              ${this._params.supervisor.localize(
+                "dialog.datadisk_move.description",
+                {
+                  current_path: this._params.supervisor.os.data_disk,
+                }
+              )}
+              <br /><br />
+
+              <paper-dropdown-menu
+                .label=${this._params.supervisor.localize(
+                  "dialog.datadisk_move.select_device"
+                )}
+                @value-changed=${this._select_device}
+              >
+                <paper-listbox slot="dropdown-content">
+                  ${devices.map(
+                    (device) =>
+                      html`<paper-item>${device.dev_path}</paper-item>`
+                  )}
+                </paper-listbox>
+              </paper-dropdown-menu>
+            `
+          : this._params.supervisor.localize("dialog.datadisk_move.no_devices")}
+
+        <mwc-button slot="secondaryAction" @click=${this.closeDialog}>
+          ${this._params?.supervisor.localize("dialog.datadisk_move.cancel")}
+        </mwc-button>
+
+        <mwc-button
+          .disabled=${!this._selected_device}
+          slot="primaryAction"
+          @click=${this._moveDatadisk}
+        >
+          ${this._params?.supervisor.localize("dialog.datadisk_move.move")}
+        </mwc-button>
+      </ha-dialog>
+    `;
+  }
+
+  private _select_device(event) {
+    this._selected_device = event.detail.value;
+  }
+
+  private async _moveDatadisk() {
+    try {
+      await moveDatadisk(this.hass, this._selected_device!);
```

Comment:
Shouldn't we close the dialog after it is done?

Reply:
When it's done the OS reboots

Reply:
Do we tell the user that?

Reply:
https://github.com/home-assistant/frontend/pull/10048/files#diff-e67939fd25c650222db710f18764d10ae69454b0e8ad680f5e10177c2db93ceaR4209

Reply:
I think we should tell them what is going to happen, and how long it will take, so they don't panic when HA goes offline, pull the plug and corrupt the fs while copying or something 😛

Reply:
Please add that, I don't know